### PR TITLE
[Dubbo-perf]: Reduce object allocation for URL.getAddress()

### PR DESCRIPTION
**Pull request by LinShunKang**
## What is the purpose of the change

Reduce object allocation for `URL.getAddress()`.

## Brief changelog

* Add `address` field for `URL.java`.
* Clean code.

## Verifying this change

XXXXX

Follow this checklist to help us incorporate your contribution quickly and easily:

- [ ] Make sure there is a [GITHUB_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GITHUB issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [x] Format the pull request title like `[Dubbo-XXX] Fix UnknownException when host config not exist #XXX`. Each commit in the pull request should have a meaningful subject line and body.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Run `mvn clean install -DskipTests=false` & `mvn clean test-compile failsafe:integration-test` to make sure unit-test and integration-test pass.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

**Comment by LinShunKang** [comment]
@chickenlj PTAL.

File: dubbo-common/src/main/java/org/apache/dubbo/common/URL.java [hunk 1]
```diff
@@ -477,7 +484,7 @@ public int getPort(int defaultPort) {
     }
 
     public String getAddress() {
-        return port <= 0 ? host : host + ":" + port;
+        return address;
```
Comment (chickenlj) [inline dubbo-common/src/main/java/org/apache/dubbo/common/URL.java]: Should we check if `address` is empty before return? Or compute address like before when address is empty?

Comment (chickenlj) [reply dubbo-common/src/main/java/org/apache/dubbo/common/URL.java]: `new URL(...)` is not the only way of allocating URL instance, if URL instance is allocated in other way, for example deserialization, the address property would be left empty.

File: dubbo-common/src/main/java/org/apache/dubbo/common/URL.java [hunk 2]
```diff
@@ -304,31 +309,33 @@ public static URL valueOf(String url) {
 
     public static Map<String, Map<String, String>> toMethodParameters(Map<String, String> parameters) {
```
[the remark this answers is not in the document]
Comment (LinShunKang) [reply dubbo-common/src/main/java/org/apache/dubbo/common/URL.java]: The nesting level of this method is too deep.

**Pull request merged**